web/src/app/ui/kmwuifloat.ts

```diff
@@ -442,15 +442,15 @@ if(!keyman?.ui?.name) {
        * @param       {Object}    e       event
        * Description  Change active keyboard in response to user selection event
        */
-      readonly SelectKeyboardChange = (e: Event) => {
+      readonly SelectKeyboardChange = async (e: Event) => {
         keymanweb.activatingUI(true);
 
         if(this.KeyboardSelector.value != '-') {
           var i=this.KeyboardSelector.selectedIndex;
           var t=this.KeyboardSelector.options[i].value.split(':');
-          keymanweb.setActiveKeyboard(t[0],t[1]);
+          await keymanweb.setActiveKeyboard(t[0],t[1]);
```

Comment:
Using the review for me learning. Just did some very quick reading of promises. So as I understand setActiveKeyboard is a promise function which will either resolve or reject. However this code doesn't take into account whether it resolved or rejected, is it just simply waiting to it has run (async) before continuing... regardless of resolved or rejected.

Reply:
> However this code doesn't take into account whether it resolved or rejected, is it just simply waiting to it has run (async) before continuing... regardless of resolved or rejected.

If rejected, an exception is thrown. Generally setActiveKeyboard should never reject.